File: shared/src/commonMain/kotlin/com/karim/foodrun/shared/orders/RoomInvitation.kt
package com.karim.foodrun.shared.orders

import com.karim.foodrun.orders.*

internal fun roomInvitation(room: Room, link: String, language: String): String {
    val arabic = language == "ar"
    val restaurant = if (room.restaurantPollOpen) {
        (if (arabic) "تصويت المطاعم: " else "Restaurant poll: ") +
            room.restaurantOptions.joinToString(" · ") { it.localizedName(language) }
    } else (if (arabic) "المطعم: " else "Restaurant: ") + room.restaurant.localizedName(language)
    return listOf("Food Run · ${room.name}", restaurant,
        (if (arabic) "طلب رقم " else "Order #") + room.orderNumber,
        (if (arabic) "كود الغرفة: " else "Room code: ") + room.code,
        if (arabic) "ادخل على طول بالرابط أو الكود، من غير موافقة." else "Join directly using the link or code. No approval needed.",
        link).joinToString("\n")
}

File: webApp/src/foodrun/roomInvitation.js
import { t, tf } from './i18n.js';

export function roomInvitation(room, link, language = 'en') {
  const name = value => language === 'ar' && value.nameAr ? value.nameAr : value.name;
  const restaurant = room.restaurantPollOpen
    ? tf('Restaurant poll: {names}', { names: (room.restaurantOptions || []).map(name).join(' · ') }, language)
    : tf('Restaurant: {name}', { name: name(room.restaurant) }, language);
  return [`Food Run · ${room.name}`, restaurant,
    tf('Order #{number}', { number: room.orderNumber }, language),
    tf('Room code: {code}', { code: room.code }, language),
    t('Join directly using the link or code. No approval needed.', language), link].join('\n');
}
